docs/error-backoff.md

```diff
@@ -0,0 +1,94 @@
+# Error Backoff
+
+Error backoff is a mechanism that throttles outgoing TChannel traffic in order to improve the
+health of the Hyperbahn cluster and its hosted services. By definition, error backoff monitors
+the status of outgoing TChannel traffic. If it determines that 1) the Hyperbahn cluster/services
+are unhealthy, and 2) the unheathy state can be improved by throttling outgoing
+traffic, error backoff will effectively throttle outgoing traffic by returning error responses
+corresponding to the out requests.
+
+There are many reasons why error backoff can help. For example, when responding to `tchannel.busy`,
+it means that a service is stressed so that requests to the service should be reduced. As another
+example, when `tchannel.declined` is received, it may indicate the absence of any healthy host
+for the service, so there is no reason to send more requests to the same service in the next short
+amount of time period.
+
+
+## Goals
+
+Error backoff works passively to defend Hyperbahn cluster and services from flooding requests and retries.
+When a service is unhealth or over stressed, there will be an increasing number of error responses.
+Such an unhealthy state should be handled by reducing the requests to the service. On the other hand,
+it is very important that error backoff never affects healthy traffic.
+
+Goals:
+
+* Monitor the health state of an Hyperbahn edge, i.e., callerName ==> serviceName. Most effectively,
+  such information should be observed locally through call responses.
+* Throttle out requests if the current unheathy state can be relieved by reducing the requests.
+* The throttling should grow if the health state continues to decline.
+* The throttling should decline if the health state improves.
+* Error backoff should converge. In other words, if the traffic stablizes in terms of failure/success
```

Comment:
This is a great point; we definitely want this; if Backoff works correctly then the server should have at most 0/1 Busy's and the client should have a high rate of "ClientSide Busy"